StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedViewRepresentable.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  EmbeddedViewRepresentable.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 1/30/25.
+//
+
+import SwiftUI
+@_spi(STP) import StripeCore
+
+struct EmbeddedViewRepresentable: UIViewRepresentable {
+    @ObservedObject var viewModel: EmbeddedPaymentElementViewModel
+
+    init(viewModel: EmbeddedPaymentElementViewModel) {
+        self.viewModel = viewModel
+        STPAnalyticsClient.sharedClient.addClass(toProductUsageIfNecessary: EmbeddedSwiftUIProduct.self)
+    }
+    
+    public func makeUIView(context: Context) -> UIView {
+        let containerView = UIView()
+        containerView.backgroundColor = .clear
+        containerView.layoutMargins = .zero
+
+        guard let embeddedPaymentElement = viewModel.embeddedPaymentElement else { return containerView }
+        embeddedPaymentElement.presentingViewController = UIWindow.visibleViewController
+
+        let paymentElementView = embeddedPaymentElement.view
+        paymentElementView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.addSubview(paymentElementView)
+
+        NSLayoutConstraint.activate([
+            paymentElementView.topAnchor.constraint(equalTo: containerView.topAnchor),
+            paymentElementView.leadingAnchor.constraint(equalTo: containerView.leadingAnchor),
+            paymentElementView.trailingAnchor.constraint(equalTo: containerView.trailingAnchor)
+        ])
+
+        return containerView
+    }
+
+    public func updateUIView(_ uiView: UIView, context: Context) {
+        // Update the presenting view controller in case it has changed
+        viewModel.embeddedPaymentElement?.presentingViewController = UIWindow.visibleViewController
+    }
+}
+
+// MARK: UIWindow and UIViewController helpers
+
+extension UIWindow {
+    static var current: UIWindow? {
```

Comment:
This one is also similar to `stp_hackilyFumbleAroundUntilYouFindAKeyWindow`

Reply:
Ah didn't see that, I tested this out in our example app and in the barebones SwiftUI app and it works!